pkg/opcfg/config.go

```diff
@@ -28,93 +29,236 @@ import (
+// webhook (if enabled).
+func GetIsControllersEnabled() bool {
+	return lookupBoolEnvVar("CONTROLLERS_ENABLED", envMustExist)
+}
+
+// GetAreControllersNamespaceScoped returns true if the controllers only watch a
+// single namespace.
+func GetAreControllersNamespaceScoped() bool {
+	scope := GetControllersScope()
+	return scope == "namespace"
+}
+
+// GetWatchNamespace returns the namespace that the operator should watch
+func GetWatchNamespace() string {
+	// The watch namespace depends on the scope of the controllers.
+	if GetAreControllersNamespaceScoped() {
+		// A namespace scoped controller only watches for objects in the namespace
+		// it is deployed in.
+		return GetOperatorNamespace()
+	}
+	// A cluster scoped controller. Return an empty string so that all namespaces
+	// are watched.
+	return ""
+}
+
+// GetControllersScope returns the scope, cluster or namespace, of the operator.
+func GetControllersScope() string {
+	return lookupStringEnvVar("CONTROLLERS_SCOPE", envMustExist)
+}
+
+// GetMetricsAddr returns the address of the manager's Prometheus endpoint. This
+// determines if its disabled or if its behind an HTTPS or HTTP scheme.
+func GetMetricsAddr() string {
+	return lookupStringEnvVar("METRICS_ADDR", envCanNotExist)
+}
+
+// GetIsProfilerEnabled returns true if the memory profiler started with the
+// manager.
+func GetIsProfilerEnabled() bool {
+	return lookupBoolEnvVar("PROFILER_ENABLED", envMustExist)
+}
+
+// GetUseCertManager returns true if cert-manager is used to setup the webhook's
+// TLS certs.
+func GetUseCertManager() bool {
+	source := lookupStringEnvVar("WEBHOOK_CERT_SOURCE", envMustExist)
+	return source == "cert-manager"
+}
+
+// GetLoggingFilePath returns the full path to the log file. If this is empty,
+// then logging will be written to the console only.
+func GetLoggingFilePath() string {
+	return lookupStringEnvVar("LOG_FILE_PATH", envCanNotExist)
+}
+
+// getLoggingMaxFileSize will return the size of the log file when writing logs
+// to a file.
+func getLoggingMaxFileSize() int {
+	return lookupIntEnvVar("LOG_MAX_FILE_SIZE", envCanNotExist)
+}
+
+// getLoggingMaxFileAge will return the age, in days, an log file stays around.
+// This only applies if writing logs to a file.
+func getLoggingMaxFileAge() int {
+	return lookupIntEnvVar("LOG_MAX_FILE_AGE", envCanNotExist)
+}
+
+// getLoggingMaxFileRotation will determine how many rotated log files it will
+// keep around. This only applies if logging to a file.
+func getLoggingMaxFileRotation() int {
+	return lookupIntEnvVar("LOG_MAX_FILE_ROTATION", envCanNotExist)
+}
+
+// GetLoggingLevel returns the logging level to use. Logging levels are: debug,
+// info, warn, error.
+func getLoggingLevel() string {
+	return lookupStringEnvVar("LOG_LEVEL", envMustExist)
+}
+
+// GetIsDebugLoggingEnabled returns true if the debug logging level is selected.
+func GetIsDebugLoggingEnabled() bool {
```

Comment:
Why do you add the `Get` prefix?

Reply:
Good point. I think since all of the other exported functions in this file start with `Get`. I kept it up. But this one makes sense to leave off though.